### PR TITLE
fix compilation when using openssl with no-deprecated

### PR DESCRIPTION
**Pull request by aldenml**
As a ref https://github.com/paullouisageneau/libdatachannel/commit/dc9a8114bc8677e97ee87a0d62a5605f3d6317ce

### CODE REVIEW

File: src/ssl.cpp [hunk 1]
```diff
@@ -175,32 +175,12 @@ namespace {
...
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-#endif
-			// openssl requires this to clean up internal structures it allocates
-			CRYPTO_cleanup_all_ex_data();
-#ifdef TORRENT_MACOS_DEPRECATED_LIBCRYPTO
-#pragma clang diagnostic pop
-#endif
-#endif
-		}
+		{}
```
Comment (arvidn) [inline src/ssl.cpp:182]: I think you can remove this destructor declaration all together now

**Comment by aldenml** [comment]
yes, it makes sense, don't merge this then, I can rebase after you merge to `master`

**Comment by arvidn** [comment]
https://travis-ci.org/github/arvidn/libtorrent/jobs/705218670#L778

**Comment by aldenml** [comment]
I see, I might need to send a PR over there, also, it fails to compile under g++9 (CC @paullouisageneau)

**Comment by arvidn** [comment]
https://ci.appveyor.com/project/arvidn/libtorrent/builds/33919607/job/xuyeoe5ujblarixv#L431

and it looks like you picked up some new code in libdatachannel too:

https://ci.appveyor.com/project/arvidn/libtorrent/builds/33919607/job/0xre06m5k2jfcwp8#L349

**Comment by paullouisageneau** [comment]
@aldenml I fixed the compilation warnings in https://github.com/paullouisageneau/libdatachannel/commit/5825e44fc85056c74d85ceb93e8e354e4ac5c00c
I think I'll the enforce the same warning level in the CI to prevent to having to fix those.

**Comment by aldenml** [comment]
@paullouisageneau great!

@arvidn I will look in to it

**Comment by paullouisageneau** [comment]
The PR https://github.com/arvidn/libtorrent/pull/4867 should fix the issues with libdatachannel.

**Comment by aldenml** [comment]
I believe that openssl pre 1.1 needs to be removed from `master` first

**Pull request closed without merging**